### ISSUE TITLE
Fails to obtain exclusivity despite no other applications using the controller and working successfully previously

### ISSUE THREAD

**Issue by oracleofire** [Ryochan7/DS4Windows#932]
I know everyone must be sick of exclusive mode issues, but this one is strange (and frustrating) to me. 

ISSUE: Exclusive mode may work fine for days, then suddenly fails to persist.

Measures have been taken to ensure nothing is beating DS4Win to the controller:
-Steam/Uplay/Nvidia services are closed/disabled
-Process Manager verifies nothing has an open handle on the DS4 when connected without DS4Win, and shows DS4Win as the sole process when DS4Win is running.

Other things I have done:
-Disabled/Re-enabled Bluetooth driver in Device Manager
-Unpaired/Re-paired controller as a device
-Uninstalled/reinstalled ViGEmBus
-Sacrificed 100 sheep upon an altar
-Voodoo

Strange behavior I've noticed is that if I close DS4Win and reopen it, connecting the controller will not give me the exclusive mode error, even though it's apparent (through not working in games) that it has indeed failed to obtain exclusivity. If I disconnect and reconnect the controller, then it will give me the error. I will always get the error until I close and reopen DS4Win, where it will have the same strange issue where it "thinks" it gets exclusivity for the first connect, but actually doesn't. 

I'm gonna include the log because it shows interesting behavior.

12/17/2019 7:09:48 PM: DS4Windows version 1.7.25
12/17/2019 7:09:48 PM: Starting...
12/17/2019 7:09:48 PM: Connection to ViGEmBus 1.16.112.0 established
12/17/2019 7:09:48 PM: Searching for controllers...
12/17/2019 7:09:48 PM: Using Exclusive Mode
12/17/2019 7:09:57 PM: Found Controller: AC:FD:93:8F:C1:5C (BT) (DS4 v.1)
12/17/2019 7:09:57 PM: Warning: Could not open DS4 AC:FD:93:8F:C1:5C exclusively. You must quit other applications like UWP apps (Netflix), Steam, Uplay, NVIDIA IN-GAME before activating the 'Hide DS4 Controller' option. For more info check https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues
12/17/2019 7:09:57 PM: Plugging in X360 Controller for input #1
12/17/2019 7:09:57 PM: Controller 1 is using Profile “Default"      ####ends on no warning
12/17/2019 7:10:13 PM: Unplugging X360 Controller for input #1      ####manual disconnect
12/17/2019 7:10:13 PM: Controller 1 was removed or lost connection
12/17/2019 7:10:22 PM: Found Controller: AC:FD:93:8F:C1:5C (BT) (DS4 v.1)
12/17/2019 7:10:22 PM: Plugging in X360 Controller for input #1
12/17/2019 7:10:22 PM: Controller 1 is using Profile “Default"
12/17/2019 7:10:22 PM: Warning: Could not open DS4 AC:FD:93:8F:C1:5C exclusively. You must quit other applications like UWP apps (Netflix), Steam, Uplay, NVIDIA IN-GAME before activating the 'Hide DS4 Controller' option. For more info check https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues   ####ends on warning

What do you make of this? Is this already a known behavior or issue? And can you make my controller work again!!? *cries* (the issue is resolved upon a restart, as many things are, but happens again after some time).

**Comment by JarnoVgr** [comment]
Not sure if this helps but are you running DS4Windows as an admin task? I've never had exclusivity issues using that method (apart when other programs are already open like Steam).

![DS4Windows_wmHBQjNAD5](https://user-images.githubusercontent.com/2751672/71094225-e5190300-21aa-11ea-9c2d-c07363ca2179.png)

**Comment by KiNGKiMO** [comment]
Same happened with me after the latest update (1.7.25).

I had to use HIDGuardian.

**Comment by oracleofire** [comment]
Thanks for the suggestion, JarnoVgr. I have it running as a task now, but the issue came up again a day after a reboot.

On the other hand, I've actually managed to discover the trigger: running Microsoft Edge. Even after closing it, DS4Win will be unable to obtain exclusive mode. Is there some sort of super stealthy hold Edge has on the controller even after being closed? Nothing shows up in the Process Manager. At least now I know how to avoid the issue (along with getting another reason to not use Edge).

**Comment by oracleofire** [comment]
I've actually discovered the issue! In my case at least, running Microsoft Edge left its process open even after closing Edge. Simply ending the task in Task Manager enables DS4Win to achieve exclusive mode again.

The wrinkle in all this was finding out WHICH process was accessing the controller; Process Manager would not show Edge. By trial and error, I found Edge to be the culprit and the solution is very simple. So if anyone has the same issue, look to Edge...
